### PR TITLE
fix(codegen): Codegens Weapons

### PR DESCRIPTION
**Pull request by mikedijkstra**
This PR codegens the remaining Weapons except for `Weapons/BaseWeapon.cs`. I didn't have time to finish that so it can be done in a follow up PR.

### CODE REVIEW

File: Projects/UOContent/Items/Weapons/BaseMeleeWeapon.cs [hunk 1]
```diff
@@ -2,16 +2,13 @@
 
 namespace Server.Items
 {
-    public abstract class BaseMeleeWeapon : BaseWeapon
+    [Serializable(0, false)]
```
Comment (kamronbatman) [inline Projects/UOContent/Items/Weapons/BaseMeleeWeapon.cs:5]: I might need to come up with some kind of codegen for this scenario. 
This class actually doesn't serialize a version variable. It passes through entirely.

File: Projects/UOContent/Items/Weapons/PoleArms/BasePoleArm.cs [hunk 2]
```diff
@@ -6,17 +6,20 @@
...
+        [SerializableFieldAttr("[CommandProperty(AccessLevel.GameMaster)]")]
+        private bool _showUsesRemaining;
 
-        private int m_UsesRemaining;
+        [SerializableField(1)]
+        [InvalidateProperties]
+        [SerializableFieldAttr("[CommandProperty(AccessLevel.GameMaster)]")]
+        private int _usesRemaining;
 
-        public BasePoleArm(int itemID) : base(itemID) => m_UsesRemaining = 150;
```
Comment (kamronbatman) [inline Projects/UOContent/Items/Weapons/PoleArms/BasePoleArm.cs:15]: This ctor was removed probably shouldn't have been.

File: Projects/UOContent/Items/Weapons/PoleArms/BasePoleArm.cs [hunk 3]
```diff
@@ -6,17 +6,20 @@
 
 namespace Server.Items
 {
-    public abstract class BasePoleArm : BaseMeleeWeapon, IUsesRemaining
+    [Serializable(0)]
```
Comment (kamronbatman) [inline Projects/UOContent/Items/Weapons/PoleArms/BasePoleArm.cs:9]: This one is not encoded so we need a false.

File: Projects/UOContent/Items/Weapons/Swords/Scimitar.cs [hunk 4]
```diff
@@ -1,15 +1,12 @@
 namespace Server.Items
 {
     [Flippable(0x13B6, 0x13B5)]
-    public class Scimitar : BaseSword
+    [Serializable(0)]
```
Comment (kamronbatman) [inline Projects/UOContent/Items/Weapons/Swords/Scimitar.cs:4]: This one should not be encoded.

File: Projects/UOContent/Items/Weapons/PoleArms/BasePoleArm.cs (hunk 2 above)
Comment (mikedijkstra) [reply Projects/UOContent/Items/Weapons/PoleArms/BasePoleArm.cs:15]: I think this is still present, it's just the diff made it look like it was removed when it was updated. See line 22 below:
```
public BasePoleArm(int itemID) : base(itemID) => _usesRemaining = 150;
```

File: Projects/UOContent/Items/Weapons/BaseMeleeWeapon.cs (hunk 1 above)
Comment (mikedijkstra) [reply Projects/UOContent/Items/Weapons/BaseMeleeWeapon.cs]: I've reverted this for now.

**Pull request merged**